File: class2relational/src/main/java/org/atlanmod/model/dynamic/relationalModel/RelationalTable.java
package org.atlanmod.model.dynamic.relationalModel;

public class RelationalTable extends RelationalElement{

    public RelationalTable(String id, String name){
        super(RelationalMetamodel.TABLE);
        super.eSetProperty("id", id);
        super.eSetProperty("name", name);
    }

    public String getId(){
        return (String) super.eGetProperty("id");
    }

    public void setId(String id){
        super.eSetProperty("id", id);
    }

    public String getName(){
        return (String) super.eGetProperty("name");
    }

    public void setName(String name){
        super.eSetProperty("name", name);
    }

    @Override
    public String toString() {
        return super.getType() +
                " (" +
                "id:" + this.getId() +
                ", name:" + this.getName() +
                ")";
    }

}

File: relational2class/src/main/scala/org/atlanmod/model/relationalmodel/RelationalTable.scala
package org.atlanmod.model.relationalmodel

import org.atlanmod.model.IdGenerator

class RelationalTable extends RelationalElement(RelationalMetamodel.TABLE) {

    def this(name: String) {
        this()
        super.eSetProperty("id", IdGenerator.id())
        super.eSetProperty("name", name)
    }

    def this(id: String, name: String) {
        this()
        super.eSetProperty("id", id)
        super.eSetProperty("name", name)
    }

    override def getId: String = super.eGetProperty("id").asInstanceOf[String]
    def getName: String = super.eGetProperty("name").asInstanceOf[String]
    def setName(name: String): Unit = super.eSetProperty("name", name)

    override def toString: String =
            super.getType + "([" + getId() + "] " + getName + ")"

}



